File: client/src/app/shared/models/blogcomment.ts
export class BlogCommentClass {

    constructor(
        public id: number,
        public blogId: number,
        public commentContent: string,
        public applicationUserId: number,
        public username: string,
        public publishedOn: Date,
        public updatedOn: Date,
        public parentBlogCommentId?: number

    ) {}
}

export class BlogCommentCreateEditClass {

    constructor(
        public id: number,
        public blogId: number,
        public commentContent: string,
        public parentBlogCommentId?: number
    ) {}
}

export class BlogCommentClientOnlyClass {

        constructor(
            public parentBlogCommentId: number,
            public blogId: number,
            public id: number,
            public commentContent: string,
            public username: string,
            public publishedOn: Date,
            public updatedOn: Date,
            public isEditable: boolean = false,
            public deleteConfirm: boolean = false,
            public isReplying: boolean = false,
            public comments: BlogCommentClientOnlyClass[]
        ) {}
    }

File: Core/Entities/Blogs/BlogComment.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace Core.Entities.Blogs
{
    public class BlogComment : BaseEntity
    {
        public int? ParentBlogCommentId { get; set; }

        public int BlogId { get; set; }
        public Blog Blog { get; set; }

        public int ApplicationUserId { get; set; }
        public ApplicationUser ApplicationUser { get; set; }

        public string CommentContent { get; set; }
        
        
        [DataType(DataType.Date)]
        public DateTime PublishedOn { get; set; }

        [DataType(DataType.Date)]
        public DateTime UpdatedOn { get; set; }



    }
}